File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.go
func productExceptSelf(nums []int) []int {
	result := []int{}

	for i := 0; i < len(nums); i++ {
		prod := 1
		for j, n := range nums {
			if j != i {
				prod *= n
			}
		}
		result = append(result, prod)
	}

	return result
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        seen = defaultdict(int)
        
        for i in range(len(nums)): 
            seen[nums[i]] += 1
            
        for k, v in seen.items(): 
            v -= 1
            if v != 0: 
                return k